v2/concordium/types.proto

```diff
@@ -462,6 +492,300 @@ message FinalizedBlockInfo {
+      }
+      // TODO: implement remaning variants
+      oneof type {
+        Transferred transferred = 1;
+      }
+    }
+    repeated Event events = 1;
+  }
+  message Reject {
+    message RejectReason {
+      Empty value = 1; // TODO: implement
+    }
+    RejectReason reject_reason = 1;
+  }
+
+  optional AccountAddress sender = 1;
+  TransactionHash hash = 2;
+  Amount cost = 3;
+  Energy energy_cost = 4;
+  oneof type {
+    AccountTransaction account = 5;
+    CredentialDeploymentTransaction credential_deployment = 6;
+    UpdateTransaction update = 7;
+  }
+  oneof result {
+    Success success = 8;
+    Reject reject = 9;
+  }
+  TransactionIndex index = 10;
+}
+
+
+
+// TODO: Consider if microgtu/euro should be the default value, as it will be used much more frequently.
+// First variant of an enum will be encoded with no bytes.
+enum UpdateType {
+    UPDATE_PROTOCOL = 0;
+    UPDATE_ELECTION_DIFFICULTY = 1;
+    UPDATE_EURO_PER_ENERGY = 2;
+    UPDATE_MICRO_GTU_PER_EURO = 3;
+    UPDATE_FOUNDATION_ACCOUNT = 4;
+    UPDATE_MINT_DISTRIBUTION = 5;
+    UPDATE_TRANSACTION_FEE_DISTRIBUTION = 6;
+    UPDATE_GAS_REWARDS = 7;
+    UPDATE_POOL_PARAMETERS = 8;
+    ADD_ANONYMITY_REVOKER = 9;
+    ADD_IDENTITY_PROVIDER = 10;
+    UPDATE_ROOT_KEYS = 11;
+    UPDATE_LEVEL1_KEYS = 12;
+    UPDATE_LEVEL2_KEYS = 13;
+    UPDATE_COOLDOWN_PARAMETERS = 14;
+    UPDATE_TIME_PARAMETERS = 15;
+}
+
+enum TransactionType {
+  DEPLOY_MODULE = 0;
+  INIT_CONTRACT = 1;
+  UPDATE = 2;
+  TRANSFER = 3;
+  // TODO: implement rest
+}
+
+// The response for getNextAccountSequenceNumber.
+message NextAccountSequenceNumber {
+  // The best guess for the available account sequence number.
+  SequenceNumber sequence_number = 1;
+  // Whether the guess relies on any non-finalized transactions. If true all of the relevant transactions are finalized.
+  bool all_final = 2;
+}
+
+// A duration of milliseconds.
+message Duration {
+  uint64 value = 1;
+}
+
+// The different versions of the protocol.
+enum ProtocolVersion {
+  PROTOCOL_VERSION_1 = 0;
+  PROTOCOL_VERSION_2 = 1;
+  PROTOCOL_VERSION_3 = 2;
+  PROTOCOL_VERSION_4 = 3;
+}
+
+// The number of chain restarts via a protocol update. An effected
+// protocol update instruction might not change the protocol version
+// specified in the previous field, but it always increments the genesis
+// index.
+message GenesisIndex {
+  uint32 value = 1;
+}
+
+// The response for GetConsensusInfo.
+message ConsensusInfo {
+  // Hash of the current best block.
+  BlockHash best_block = 1;
+  // Hash of the (original) genesis block.
+  BlockHash genesis_block = 2;
+  // Time of the (original) genesis block.
+  Timestamp genesis_time = 3;
+  // (Current) slot duration in milliseconds.
+  Duration slot_duration = 4;
+  // (Current) epoch duration in milliseconds.
+  Duration epoch_duration = 5;
+  // Hash of the last finalized block.
+  BlockHash last_finalized_block = 6;
+  // Absolute height of the best block.
```

Comment:
Consider mentioning that this means from the original genesis block. Also for the field below.

Reply:
I find it a bit noisy to explain the definition of absolute height everywhere we use it and I would prefer to only have it next to the type definition.